File: leetcode/1544/make_the_string_great.cpp
// 0ms | 7.5mb
// 100.0% | 98.66%
class Solution {

public:
  string makeGood(string s) {
    for (auto si = s.begin() + 1; si < s.end(); ++si) {
      if (tolower(*(si - 1)) == tolower(*(si)) && *(si - 1) != *(si)) {
        s.erase(si);
        s.erase(si - 1);
        si = s.begin();
      }
    }
    return s;
  }
};

File: leetcode/1544/make_the_string_great.py
# 30ms | 16.47mb
# 93.52% | 91.17%
class Solution:
    def makeGood(self, s: str) -> str:
        pointer = 1
        while pointer < len(s):
            if s[pointer].lower() == s[pointer-1].lower() and s[pointer] != s[pointer-1]:
                s = s[:pointer-1] + s[pointer+1:]
                pointer = max(pointer - 2, 0)
            pointer += 1
        return s
